### PR TITLE
Add Adaptive Links / Clean Up Duplicates / Dead Links

### PR DESCRIPTION
**Pull request by MapGuy11**

### CODE REVIEW

File: channels/us.m3u [hunk 1]
```diff
@@ -1,8 +1,6 @@
 #EXTM3U
 #EXTINF:-1 tvg-id="KOTR-LD.us" tvg-name="11 MyTV (KOTR-LD)" tvg-country="US" tvg-language="English" tvg-logo="https://upload.wikimedia.org/wikipedia/en/b/bd/Kotrlp.jpg" group-title="Local",11 MyTV (KOTR-LD) (480i)
 https://cdnapisec.kaltura.com/p/2630451/sp/2630451/playManifest/entryId/1_hq4vlk40/deliveryProfileId/672/protocol/https/format/applehttp/a.m3u8
-#EXTINF:-1 tvg-id="13thStreet.us" tvg-name="13th Street (Poland)" tvg-country="PL" tvg-language="Polish" tvg-logo="" group-title="",13th Street (Poland)
-http://91.218.60.3:8001/play/a004
```
Comment (Dum4G) [inline channels/us.m3u:5]: image

**Review by Dum4G — CHANGES_REQUESTED** [review]

File: channels/us.m3u [hunk 2]
```diff
@@ -69,22 +63,16 @@ https://streams.the6tv.duckdns.org:2443/locals/BayArea/kgo-dt1.m3u8
...
 #EXTINF:-1 tvg-id="AdultSwim.us" tvg-name="Adult Swim" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/ixywxhs.jpg" group-title="Entertainment",Adult Swim (1080p)
 https://adultswimtv.warnermediacdn.com/hls/live/2007031/livestream/master.m3u8
 #EXTINF:-1 tvg-id="AdventureSportsNetwork.us" tvg-name="Adventure Sports Network" tvg-country="US" tvg-language="English" tvg-logo="" group-title="Sports",Adventure Sports Network
-https://dai2.xumo.com/amagi_hls_data_xumo1212A-adventuresportsnetwork/CDN/768x432_1200000/index.m3u8
-#EXTINF:-1 tvg-id="AdventureSportsNetwork.us" tvg-name="Adventure Sports Network" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/ZbPFZOV.png" group-title="Sports",Adventure Sports Network (XUMO)
-https://dai2-xumo.fubo.tv/amagi_hls_data_xumo1212A-adventuresportsnetwork/CDN/master.m3u8
-#EXTINF:-1 tvg-id="AdventureSportsTV.us" tvg-name="Adventure Sports TV" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/HL7fwzt.png" group-title="Sports",Adventure Sports TV (360p)
-https://gizmeon.s.llnwi.net/channellivev3/live/master.m3u8?channel=275
```
Comment (Dum4G) [inline channels/us.m3u:87]: It's fine to have multiple CDNs to choose from, isn't it? https://gizmeon.s.llnwi.net/channellivev3/live/master.m3u8?channel=275 can be a backup one

**Review by Dum4G — CHANGES_REQUESTED** [review]

File: channels/us.m3u [hunk 3]
```diff
@@ -121,8 +101,6 @@ https://dai.google.com/linear/hls/event/4RH6FntvSLOIv5FB-p4I8w/master.m3u8
 https://dai.google.com/linear/hls/event/-V3XSvA2Sa6e8h7cnHXB8w/master.m3u8
 #EXTINF:-1 tvg-id="WRGB.us" tvg-name="Albany | CBS 6 WRGB" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/ByPFQGp.png" group-title="Local",Albany | CBS 6 WRGB (720p)
 https://content.uplynk.com/channel/bba3e7da884a49bba96341ecf5128f0f.m3u8
-#EXTINF:-1 tvg-id="Alhurra.us" tvg-name="Alhurra" tvg-country="US" tvg-language="" tvg-logo="" group-title="",Alhurra
-http://mbnhls-lh.akamaihd.net/i/MBN_1@118619/master.m3u8?v=1558393494
 #EXTINF:-1 tvg-id="AlientoVision.us" tvg-name="Aliento Vision" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/0qwmBEm.png" group-title="Religious",Aliento Vision
 http://livestreamcdn.net:1935/AlientoSD/smil:AlientoSD.smil/playlist.m3u8
```
Comment (Dum4G) [inline channels/us.m3u:105]: Better use
https://livestreamcdn.net:444/Alientoenvivo/Alientoenvivo.smil/playlist.m3u8

File: channels/us.m3u (hunk 1 above)
Comment (MapGuy11) [reply channels/us.m3u:5]: But it isn't US that is why I removed it

**Review by Dum4G — CHANGES_REQUESTED** [review]

File: channels/us.m3u (hunk 1 above)
Comment (Dum4G) [reply channels/us.m3u:5]: We're following Lyngsat principles...
https://www.lyngsat.com/tvchannels/us/13-Ulica.html

Comment (MapGuy11) [reply channels/us.m3u:5]: Okay, but just to let you know, we can't get East Satellite in the US and only go down to 60s ish in the Western Hemisphere.

File: channels/us.m3u [hunk 4]
```diff
@@ -104,7 +104,7 @@ https://dai.google.com/linear/hls/event/-V3XSvA2Sa6e8h7cnHXB8w/master.m3u8
 #EXTINF:-1 tvg-id="WRGB.us" tvg-name="Albany | CBS 6 WRGB" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/ByPFQGp.png" group-title="Local",Albany | CBS 6 WRGB (720p)
 https://content.uplynk.com/channel/bba3e7da884a49bba96341ecf5128f0f.m3u8
 #EXTINF:-1 tvg-id="AlientoVision.us" tvg-name="Aliento Vision" tvg-country="US" tvg-language="English" tvg-logo="https://i.imgur.com/0qwmBEm.png" group-title="Religious",Aliento Vision
```
Comment (Dum4G) [inline channels/us.m3u:108]: We should clarify that it's broadcasting in Spanish

**Review by Dum4G — CHANGES_REQUESTED** [review]

**Comment by LaneSh4d0w** [comment]
Won't be easy if we're both doing it lol

**Comment by MapGuy11** [comment]
> Won't be easy if we're both doing it lol

I was just quickly doing it. Can you merge it says there is a conflict that doesn't make any sense?

**Comment by freearhey** [comment]
The conflict is resolved, but I haven't checked the rest.

**Comment by freearhey** [comment]
@MapGuy11 Why did you decide to remove all but one link to the Cheddar channel?

**Comment by MapGuy11** [comment]
> @MapGuy11 Why did you decide to remove all but one link to the Cheddar channel?

I added another one. 
1. Most of them were Test Cards with their old logo or Dead Urls (They Were Bought Out By A Big Cable Provider in the States; Altice USA)
2. Cheddar Big News Died A Couple of Years Ago (Also Their Cheddar U too; Low Viewership and it was kinda the same content just at a different time)
3. XUMO has issues when you block their AD Beacon (A lot of people in the states use Whole Home AD Blockers)
4. Pluto TV Has Issues when it goes to a commercial break (FFMPEG, Plex, Other Apps have issues. And a lot of people in the states use Whole Home AD Blockers, etc.)

**Comment by freearhey** [comment]
@MapGuy11 I understand the first point, but as for other three I think I should explain something. 

The links in the `channels/us.m3u` file are not just for the US, they only get into this file based on what country they are broadcast from. So if they work in most of the world then there is no reason to remove them.

Also if the channel has some shift in the program you can leave a link to it and just add the time shift in the name.

And lastly, about a month ago I added to the script a search for duplicates by their description. Now if two links have exactly the same description only the first of them will get into the public playlist. So it will be enough to edit the channel description so that the same channel is not shown in the playlist twice.

**Pull request closed without merging**